src/main/java/zmq/io/mechanism/Mechanisms.java

```diff
@@ -36,6 +53,22 @@ public Mechanism create(SessionBase session, Address peerAddress, Options option
         }
     },
     CURVE {
+        @Override
+        public void check(Options options)
+        {
+            if ((options.curvePublicKey == null ^ options.curveSecretKey == null)) {
+                throw new IllegalStateException("Curve mechanism definition incomplete");
+            }
+            if (options.curvePublicKey != null && options.curvePublicKey.length != Options.CURVE_KEYSIZE) {
+                throw new IllegalStateException("Curve mechanism definition incomplete");
+            }
+            if (options.curveSecretKey != null && options.curveSecretKey.length != Options.CURVE_KEYSIZE) {
```

Comment:
As a user, I would be welcoming all options config problems at once, the proposed algo means that if 2 keys are misconfigured, you will have to try 2 times to find out all the problems.